### PR TITLE
sniffer: use argparse to parse script arguments

### PR DESCRIPTION
**Pull request by miri64**
This is mainly a preparation step to make some arguments (namely
baudrate and the "serial/socket" argument) optional, to make it more
usable like the old `rftest2pcap` script for the econotag.

(it also makes the script code more readable)

### CODE REVIEW

File: sniffer/tools/sniffer.py [hunk 1]
```diff
@@ -135,25 +136,39 @@ def connect(argv):
...
+    socket_p = sp.add_parser("socket",
+                             help="Parse output of a TCP-connected sniffer "
+                                  "application")
+    socket_p.add_argument("host", type=str,
+                          help="Host of the TCP-based sniffer application")
+    socket_p.add_argument("port", type=int,
+                          help="Port of the TCP-based sniffer application")
+    p.add_argument("channel", type=int, help="Channel to sniff on")
+    p.add_argument("outfile", type=str, default=None, nargs="?",
+                   help="PCAP file to output to")
```
Comment (bergzand) [inline sniffer/tools/sniffer.py]: There is an [argparse.FileType](https://docs.python.org/3.5/library/argparse.html#argparse.FileType) in python.

Comment (miri64) [reply sniffer/tools/sniffer.py]: Will check it out ;-)

Comment (miri64) [reply sniffer/tools/sniffer.py]: Done

**Comment by smlng** [comment]
please squash

**Comment by miri64** [comment]
Squashed

**Pull request merged**